app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsFragment.kt

```diff
@@ -243,19 +291,27 @@ class SessionDetailsFragment : Fragment(), SessionDetailsViewModel.ViewActionHan
         this.isVisible = true
     }
 
+    private fun updateOptionsMenu() {
+        requireActivity().invalidateOptionsMenu()
+    }
+
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
+        if (!::model.isInitialized) {
+            // Skip if lifecycle is faster than ViewModel.
+            return
+        }
         inflater.inflate(R.menu.detailmenu, menu)
-        if (viewModel.isFlaggedAsFavorite) {
+        if (model.isFlaggedAsFavorite) {
             menu.findItem(R.id.menu_item_flag_as_favorite)?.let { it.isVisible = false }
             menu.findItem(R.id.menu_item_unflag_as_favorite)?.let { it.isVisible = true }
         }
-        if (viewModel.hasAlarm()) {
+        if (model.hasAlarm) {
             menu.findItem(R.id.menu_item_set_alarm)?.let { it.isVisible = false }
             menu.findItem(R.id.menu_item_delete_alarm)?.let { it.isVisible = true }
         }
-        var item: MenuItem? = menu.findItem(R.id.menu_item_feedback)
-        if (SHOW_FEEDBACK_MENU_ITEM && !viewModel.isFeedbackUrlEmpty) {
+        var item = menu.findItem(R.id.menu_item_feedback)
+        if (SHOW_FEEDBACK_MENU_ITEM && !model.isFeedbackUrlEmpty) {
             item?.let { it.isVisible = true }
         } else {
             item?.let { it.isVisible = false }
```

Comment:
And with the extension you could for example just do:
```
item.setMenuItemVisibility(
  R.id.menu_item_feedback,
  SHOW_FEEDBACK_MENU_ITEM && !model.isFeedbackUrlEmpty
)
```

Reply:
:heavy_check_mark: Adapted in 705739fa159e2ce83615bf013a493b145b39d48a.